hazelcast-hibernate5/src/main/java/com/hazelcast/hibernate/AbstractHazelcastCacheRegionFactory.java

```diff
@@ -123,4 +125,14 @@ public HazelcastInstance getHazelcastInstance() {
     public AccessType getDefaultAccessType() {
         return AccessType.READ_WRITE;
     }
+
+    public void setPhoneHomeService(PhoneHomeService phoneHomeService) {
```

Comment:
Maybe let's just inject `PhoneHomeService` via constructor? The field could be _final_ and a setter could be removed which would result in fewer moving parts.

Reply:
The setter stands only for test mocks since the service is initialized in-place. May be I should've indicate within a comment.

Reply:
I got that, but tests should not influence the production code. And it feels like raw dependency injection(with my favourite dependency injection framework being _constructors_!) is a solution to the problem where dependencies are supplied from outside and not created internally